src/indexes.rs

```diff
@@ -466,6 +466,30 @@ impl Index {
         &self,
         documents_query: &DocumentsQuery<'_>,
     ) -> Result<DocumentsResults<T>, Error> {
+        if documents_query.filter.is_some() {
+            let url = format!("{}/indexes/{}/documents/fetch", self.client.host, self.uid);
+            return request::<(), &DocumentsQuery, DocumentsResults<T>>(
+                &url,
+                self.client.get_api_key(),
+                Method::Post {
+                    body: documents_query,
+                    query: (),
+                },
+                200,
+            )
+            .await
+            .map_err(|err| match err {
+                Error::MeilisearchCommunication(error) => {
+                    Error::MeilisearchCommunication(MeilisearchCommunicationError {
+                        status_code: error.status_code,
+                        url: error.url,
+                        message: Some("It might not be working because you're not up to date with the Meilisearch version that updated the get_documents_with method.".to_string()),
```

Comment:
> Wouldn't it be better to create a way to add a message directly to an existing error? I'm not sure how to do it in rust.

There is no initial message with isahc on a server returning a HTTP code above 400. So for example, 405 has no message with `Method Not Allowed` written.

> but if in the next update of Meilisearch, it is possible to add other fields in addition to filters, then the error may be a MeilisearchAPI error right?


Not sure I understand this part. Do you have an example?

In case this answers your question:
The communication error is triggered when:
- `status_code >= 400 `
- it was not able to parse the error into a MeilisearchApiError
- it was able to reach Meilisearch (in contrario of a httpError) 

So if Meilisearch returns an error with `unknown field` because in the future we add new params on this route, then, there is no hint. There are a lot of reason that might cause an error and I rather not confuse the user with hints if Meilisearch already provides an error message.

Reply:
After discussing it in private, we decided that the hint should only be provided when there is a communication error with meilisearch